extensions-support/httpclient5/deployment/src/main/java/org/apache/camel/quarkus/support/httpclient5/deployment/HttpClient5Processor.java

```diff
@@ -34,4 +46,54 @@ NativeImageResourceBuildItem suffixListResource() {
     void runtimeInitializedClasses(BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitializedClasses) {
         runtimeInitializedClasses.produce(new RuntimeInitializedClassBuildItem(NTLM_ENGINE_IMPL));
     }
+
+    @BuildStep(onlyIf = { NativeOrNativeSourcesBuild.class, BrotliAbsentBooleanSupplier.class })
+    void generateBrotliInputStreamClass(BuildProducer<GeneratedClassBuildItem> generatedClass) {
+        try (ClassCreator classCreator = ClassCreator.builder()
+                .className(BROTLI_INPUT_STREAM_CLASS_NAME)
+                .superClass(InputStream.class)
+                .classOutput(new GeneratedClassGizmoAdaptor(generatedClass, false))
+                .build()) {
+
```

Comment:
Maybe comments below could be removed.

Reply:
I deliberately left them in so that it's obvious what the generated class source will look like.